utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/source/UniversalKafkaSourceFactory.scala

```diff
@@ -229,10 +229,12 @@ class UniversalKafkaSourceFactory(
 
 object UniversalKafkaSourceFactory {
 
-  case class UniversalKafkaSourceFactoryState(
+  trait UniversalKafkaSourceFactoryState
```

Comment:
Why this common trait for both cases? And what does mean this Final prefix? Which one state is non final?